src/actions/arcActions.ts

```diff
@@ -267,30 +277,17 @@ async function getProposalDetails(daoInstance: Arc.DAO, votingMachineInstance: A
 
   const submittedTime = Number(proposalDetails.submittedTime);
 
-  // Use title and description from the server
-  let description = "";
+  // Use title and url from the ipfs server
+  let url = "";
   let title = "[no title]";
   if (serverProposal) {
-    description = serverProposal.description;
+    url = serverProposal.url;
     title = serverProposal.title;
   } else {
-    // If we didn't find the proposal by proposalId, see if there is one that matches by description hash that doesnt yet have a proposalId added to it
-    let response = await axios.get(process.env.API_URL + '/api/proposals?filter={"where":{"and":[{"arcId":null},{"daoAvatarAddress":"' + avatarAddress + '"},{"descriptionHash":"' + descriptionHash + '"}]}}');
-    if (response.data.length > 0) {
-      serverProposal = response.data[0];
-      description = serverProposal.description;
-      title = serverProposal.title;
-
-      // If we found one, then update the database with the proposalId
-      response = await axios.patch(process.env.API_URL + '/api/proposals/' + serverProposal.id, {
-        arcId: proposalId,
-        daoAvatarAddress: avatarAddress,
-        descriptionHash,
-        description,
-        submittedAt: submittedTime,
-        title
-      });
-    }
+    let ipfsHash = ipfs.hexToHash(descriptionHash)
+    let serverProposal = await ipfs.get(ipfsHash)
+    url = serverProposal.url;
```

Comment:
This can and will go wrong - hash not found, incorrect document found under ipfsHash . We need some error handling here.

Reply:
Good catch, will look into handling edge cases better here